lib/pure/strutils.nim

```diff
@@ -154,6 +154,61 @@ proc isDigit*(c: char): bool {.noSideEffect,
     doAssert isDigit('8') == true
   return c in Digits
 
+proc isNumeric*(s: string): bool =
+  ## Checks whether the string is numeric.
+  ## When the string is an integer, float or exponential, it returns true,
+  ## otherwise it returns false.
+  ##
+  ## **Note:** The reason that `parseFloat()` is not used to achieve this is its
+  ## poor performance.
+  runnableExamples:
+    doAssert isNumeric("123") == true
+    doAssert isNumeric("123.45") == true
+    doAssert isNumeric("+123.45") == true
+    doAssert isNumeric("123.45e-2") == true
+    doAssert isNumeric("+123.45E-2") == true
+    doAssert isNumeric("-123.45e2") == true
+    doAssert isNumeric("e123.45") == false
+    doAssert isNumeric("abc") == false
+    doAssert isNumeric("123abc") == false
+    doAssert isNumeric("123.45.6") == false
+    doAssert isNumeric("123.45e++5") == false
+    doAssert isNumeric(".9") == true
+    doAssert isNumeric("Inf") == true
+    doAssert isNumeric("-Inf") == true
+    doAssert isNumeric("NaN") == true
+  if s == "Inf" or s == "inf" or s == "-Inf" or s == "-inf" or s == "NaN" or s == "nan": return true
```

Comment:
You should use `toLowerAscii` instead

```nim
case toLowerAscii(s)
of "inf", "nan", "-inf":
```
Or you should check in while loop

Reply:
I have considered using `toLowerAscii()`, but its performance loss is too great.

Reply:
https://play.nim-lang.org/#ix=2DJt

Reply:
Your if statement is wrong.
NAn or nAN should be considered too.

Reply:
you could check it in while loop for performance reason.

Reply:
Other way is to use something like `strict: static bool = false` then you can use `when strict:`...

Reply:
I don't think `static` makes a difference in practice because of branch prediction; at least a test is needed if you think otherwise